src/Transport/Topology/MetaModel/EntityInfo.cs

```diff
@@ -23,17 +22,12 @@ public EntityInfo()
 
         protected bool Equals(EntityInfo other)
         {
-            if (Type == EntityType.Subscription)
-            {
-                var entity = RelationShips.First(r => r.Type == EntityRelationShipType.Subscription);
-                var otherEntity = other.RelationShips.First(r => r.Type == EntityRelationShipType.Subscription);
-                var targetPathEquals = string.Equals(entity.Target.Path, otherEntity.Target.Path);
-                if (!targetPathEquals)
-                {
-                    return false;
-                }
-            }
-            return string.Equals(Path, other.Path) && Type == other.Type && Equals(Namespace, other.Namespace);
+            return string.Equals(Path, other.Path) && Type == other.Type && Equals(Namespace, other.Namespace) && DerivedEqual(other);
```

Comment:
@Particular/azure-maintainers can we please make a decision on this?
This is what API approval yields if `Equals(object)` and `GetHashCode()` in `SubscriptionInfo` is overridden:
![image](https://cloud.githubusercontent.com/assets/1309622/21624932/7afd494a-d1c6-11e6-92ac-e4ee8f045d3e.png)

So this could be a bug in APIApproval, not sure. Even if it's not an issue, overriding `GetHashCode()` for `SubscriptoinInfo` is now changing the method`s behavior as it's no longer going to be the one of the base class.